graphium/data/datamodule.py

```diff
@@ -766,6 +767,7 @@ def __init__(
         featurization_progress: bool = False,
         featurization_backend: str = "loky",
         featurization_batch_size: int = 1000,
+        sampler_task_dict: Optional[dict] = None,
```

Comment:
The value of `sampler_task_dict` is set to `Optional`, but you don't handle the case where it is `None`.

Also, as mentioned above, it's better if it's in the `task_specific_args`.
This means that the ratios will be assigned to the `SingleTaskDataset`, and then you can create the `sampler_task_dict` from the `MultiTaskDataset`. If a value is missing, the correct default should be `1.0`, where you take the full dataset.

Reply:
Cool, I've made the changes accordingly. The only thing I could think of is that this approach would need the cache to be regenerated each time we change the `epoch_sampling_fraction`, even though the cache does not actually need to be regenerated. Let me know your thoughts.

Reply:
You can exclude the `epoch_sampling_fraction` from the hash key. I `get_data_hash` you can pop it from `task_specific_args`

Reply:
good point!